lib/net.js

```diff
@@ -371,8 +371,8 @@ Socket.prototype._final = function(cb) {
 };
 
 
-function afterShutdown(status, handle) {
```

Comment:
Do you know why this gets called with the wrong `handle`?

Reply:
It comes from here: https://github.com/nodejs/node/blob/v11.4.0/src/stream_base.cc#L383
This in turn is set by https://github.com/nodejs/node/blob/v11.4.0/src/stream_base-inl.h#L86
which is called by https://github.com/nodejs/node/blob/v11.4.0/src/tls_wrap.cc#L73

`TLSWrap`'s `stream` is the external stream from here: https://github.com/nodejs/node/blob/v11.4.0/lib/_tls_wrap.js#L423

which is the unencrypted socket passed into the constructor: https://github.com/nodejs/node/blob/v11.4.0/lib/_tls_wrap.js#311

In summary, it gets the underlying socket's stream rather than the encrypted stream (`TLSSocket`). Node stream operations appear to be performed on `TLSSocket` only.

Reply:
Thanks for the explanation.